app/src/main/java/edu/uw/covidsafe/importlocations/ImportGoogleLocations.java

```diff
@@ -0,0 +1,165 @@
+import android.util.Log;
+import android.util.Xml;
+import android.view.View;
+import android.webkit.CookieManager;
+import android.webkit.DownloadListener;
+import android.webkit.URLUtil;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.System.in;
+
+public class MainActivity extends AppCompatActivity {
+
+    private static final String TAG="AndroidRide";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        final Context context = getApplicationContext();
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Takeout page: https://takeout.google.com/settings/takeout/custom/location_history
+        // KML page: "https://www.google.com/maps/timeline/kml?authuser=0&pb=!1m8!1m3!1i" + year + "!2i" + month + "!3i" + mday +
+        //        "!2m3!1i" + year + "!2i" + month + "!3i" + mday;
+        // Ex: "https://www.google.com/maps/timeline/kml?authuser=0&pb=!1m8!1m3!1i" + 2020 + "!2i" + 01 + "!3i" + 01 +
+        //                "!2m3!1i" + 2020 + "!2i" + 04 + "!3i" + 01
+        final String signin_url = "https://accounts.google.com/signin/v2/identifier?flowName=GlifWebSignIn&flowEntry=ServiceLogin";
+        final String download_url = "https://www.google.com/maps/timeline/kml?authuser=0&pb=!1m8!1m3!1i" + 2020 + "!2i" + 01 + "!3i" + 01 +
+                "!2m3!1i" + 2020 + "!2i" + 01 + "!3i" + 04;
+        final Map<String, String> headers = new HashMap<>();
+        final WebView myWebView = (WebView) findViewById(R.id.webview);
+
+        headers.put("content-type", "application/vnd.google-earth.kml+xml");
+        WebSettings webSettings = myWebView.getSettings();
+        webSettings.setJavaScriptEnabled(true);
+
+        myWebView.setWebViewClient(new MyWebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                super.onPageFinished(myWebView, url);
+                if (url.contains("https://myaccount.google.com/")) {
+                    checkStoragePermission();
+                    myWebView.loadUrl(download_url, headers);
+                    myWebView.setVisibility(View.GONE);
+                    myWebView.clearFormData();
+                    myWebView.clearCache(true);
+                    try {
+                        Thread.sleep(5000);

```

Comment:
Hey @aadam3541 , We should not make the main thread sleep, it will lead to application freeze, any specific reasoning for doing this?